### PR TITLE
[orchestra] fix: require the initialization with `F: FnOnce` to be `Send`

### PR DESCRIPTION
**Pull request by drahnr**
If creating intermediate variables of the builder type within a future, rustc will complain about the future not being send, while the thing itself isn't even using the closure based field initialization. Adding an additional bound, resolves this and pushes the error message "closer" to the user, and to the public interface boundary of the generated code.

CC @sandreim 

Polkadot address: 1TmDZDCaYhbE3Uip8n6GxBbfFYpikxppTp18qJFboSKPY1U

### CODE REVIEW

File: node/orchestra/examples/solo.rs [hunk 1]
```diff
@@ -46,24 +46,31 @@ impl<Context> Fortified {
 	}
 }
 
-fn main() {
-	use futures::{executor, pin_mut};
+async fn setup() {
+	let builder = Solo::builder();
```
Comment (sandreim) [inline node/orchestra/examples/solo.rs:50]: We need to add `use futures::pin_mut` so we test build doesn't fail:
```
error: cannot find macro `pin_mut` in this scope
  --> node/orchestra/examples/solo.rs:63:2
   |
63 |     pin_mut!(orchestra_fut);
   |     ^^^^^^^
  
```

File: node/orchestra/examples/solo.rs [hunk 2]
```diff
@@ -46,24 +46,31 @@ impl<Context> Fortified {
...
-		pin_mut!(orchestra_fut);
+	pin_mut!(orchestra_fut);
+
+	orchestra_fut.await;
+}
+
+fn assert_t_impl_trait_send<T: Send>(_: &T) {}
+
+fn main() {
+	use futures::{executor, pin_mut};
```
Comment (sandreim) [inline node/orchestra/examples/solo.rs:71]:
```suggestion
	use futures::executor;
```

**Review by sandreim — APPROVED** [review]
Apart from build issues, LGTM!

**Review by eskimor — APPROVED** [review]

**Comment by ordian** [comment]
bot merge

**Comment by drahnr** [comment]
:wave: @eskimor - you'll see me around `orchestra` a bit more in the next weeks/months :waffle:

**Pull request merged**